collection.go

```diff
@@ -584,11 +585,22 @@ func resolveKconfig(m *MapSpec) error {
 		return errors.New("map value is not a Datasec")
 	}
 
+	scanner, err := kconfig.Find()
+	if err != nil {
+		return fmt.Errorf("cannot find a kconfig file: %w", err)
+	}
+
+	config, err := kconfig.Parse(scanner)
```

Comment:
I think Timo is right, it would be nice if we only parsed the kconfig when required. 
I've updated the benchmark you added to use a bigger file and looked at the results very briefly: it takes around 1ms on my machine.

What if you did the following?
* in the default switch case collect the necessary config items into `map[string][]byte` or similar
* after we've iterated the items once, parse kconfig `if len(map[string][]byte) > 0` and fill in the values

That should be doable without too many code changes.

Coming back to the benchmark, most of the time is spent constructing the full `map[string]string`. That's fine for now, since it's nice and simple. But later on we could experiment using the `map[string][]byte` to only parse the lines we're actually interested in.

Reply:
Makes sense, sorry if I didn't see that. I think we might also be able to cut down memory usage to a point where we can afford to keep a cached map around.